### DISCUSSION TITLE
Cert error in AWS lambda for .net core 8 SSL_CERT_DIR

### DISCUSSION THREAD

**Discussion by sajanmje** [aws/aws-lambda-dotnet#1735] [General]
We're attempting to retrieve a list of secrets from the AWS Parameter Store in .NET Core 8. To enable external API calls, we included a certificate in the Lambda's environment variables with the key SSL_CERT_DIR and the value /opt/certs. However, after adding this variable, we encountered a certificate error while retrieving data from the parameter store. It worked fine without the SSL_CERT_DIR environment variable. How can we resolve this issue?

**Comment by normj** [comment]
Can you try setting `SSL_CERT_FILE` to the certificate you are downloading from parameter store?

With .NET 8 and the switch to Amazon Linux 2023 we found significant performance regression because `SSL_CERT_FILE` and `SSL_CERT_DIR` were basically pointing to the same location by default causing a double load of certs affecting the cold start. [To work around this we set](https://github.com/aws/aws-lambda-dotnet/blob/master/Libraries/src/Amazon.Lambda.RuntimeSupport/bootstrap-al2023.sh#L11C10-L11C23) `SSL_CERT_FILE` to a dummy file to avoid the double load. I suspect what is happening in your case since you are setting `SSL_CERT_DIR` to your file you are then missing the default certs.
